### PR TITLE
Release/2.3.0

### PR DESCRIPTION
**Pull request by JesusMcCloud**
* Kotlin 2.0.20 (binary-incompatible change, makes iOS Tests fail too)
  * Serialization 1.7.2
  * Antlr-Kotlin 1.0.0-RC5
  * add tests
  * auto-generate Kotlin sources from Antlr-files on every Gradle invocation

### CODE REVIEW

**Comment by JesusMcCloud** [comment]
we need to wait for a new Kotest release. don't know why but simply using a snapshot version is not enough for this project (in contrast to Signum, for example)

File: gradle.properties [hunk 1]
```diff
@@ -16,6 +16,6 @@ kotlin.native.ignoreDisabledTargets=true
 # workaround dokka bug (need to wait for next snapshot build)
 org.jetbrains.dokka.classpath.excludePlatformDependencyFiles=true
 
-artifactVersion = 2.2.0
+artifactVersion = 2.3.0-SNAPSHOT
```
Comment (acrusage-iaik) [inline gradle.properties:19]: Publication versions should not be labeled "SNAPSHOT"

File: jsonpath4k/build.gradle.kts [hunk 2]
```diff
@@ -43,8 +61,9 @@ val generateKotlinGrammarSource = tasks.register<AntlrKotlinTask>("generateKotli
     arguments = listOf("-visitor")
 
     // Generated files are output inside build/generatedAntlr/{package-name}
```
Comment (acrusage-iaik) [inline jsonpath4k/build.gradle.kts:63]: This comment is outdated

File: settings.gradle.kts [hunk 3]
```diff
@@ -6,6 +6,7 @@ pluginManagement {
         google()
         gradlePluginPortal()
         mavenCentral()
+        maven("https://s01.oss.sonatype.org/content/repositories/snapshots") //KOTEST snapshot
```
Comment (acrusage-iaik) [inline settings.gradle.kts:9]: I'm not a fan of allowing snapshot repositories for releases, but if this can't be circumvented then so be it

**Review by acrusage-iaik — CHANGES_REQUESTED** [review]
minor changes

File: gradle.properties (hunk 1 above)
Comment (JesusMcCloud) [reply gradle.properties]: fixed

File: jsonpath4k/build.gradle.kts (hunk 2 above)
Comment (JesusMcCloud) [reply jsonpath4k/build.gradle.kts]: fixed

File: settings.gradle.kts (hunk 3 above)
Comment (JesusMcCloud) [reply settings.gradle.kts]: nope. not needed, but iOS tests are failing

**Review by acrusage-iaik — APPROVED** [review]

**Pull request merged**